internal/pkg/action/operator_list_custom_resources.go

```diff
@@ -0,0 +1,113 @@
+	Namespace string
+}
+
+func NewOperatorListCustomResources(cfg *Configuration) *OperatorListCustomResources {
+	return &OperatorListCustomResources{
+		config: cfg,
+		Namespace: cfg.Namespace,
+	}
+}
+
+// FindOperator finds an operator object on-cluster provided a package and namespace. Returns a copy of the operator.
+func (o *OperatorListCustomResources) FindOperator(ctx context.Context) (v2alpha1.Operator, error) {
+	opKey := types.NamespacedName{
+		Name:      o.PackageName,
+		Namespace: o.Namespace,
+	}
+	operator := v2alpha1.Operator{}
+	err := o.config.Client.Get(ctx, opKey, &operator)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return v2alpha1.Operator{}, fmt.Errorf("could not find operator for package %s in namespace %s", o.PackageName, o.Namespace)
+		}
+		return v2alpha1.Operator{}, err
+	}
+	return operator, nil
+}
+
+// Unzip finds the CSV referenced by the provided operator and then inspects the spec.customresourcedefinitions.owned
+// section of the CSV to return a list of APIs that are owned by the CSV.
+func (o *OperatorListCustomResources) Unzip(ctx context.Context, operator *v2alpha1.Operator) ([]v1alpha1.CRDDescription, error) {
+	csv := v1alpha1.ClusterServiceVersion{}
+	csvKey := types.NamespacedName{}
+
+	if operator.Status.Components == nil {
+		return nil, fmt.Errorf("could find underlying components for operator %s", operator.Name)
+	}
+	for _, resource := range operator.Status.Components.Refs {
+		if resource.Kind == v1alpha1.ClusterServiceVersionKind {
+			csvKey.Name = resource.Name
+			csvKey.Namespace = resource.Namespace
+			break
+		}
+		return nil, fmt.Errorf("could find underlying CSV for operator %s", operator.Name)
+	}
+
+	err := o.config.Client.Get(ctx, csvKey, &csv)
+	if err != nil {
+		return nil, fmt.Errorf("could not get %s CSV on cluster: %s", csvKey.String(), err)
+	}
+
+	// check if owned CRDs are defined on the csv first
+	if reflect.DeepEqual(csv.Spec.CustomResourceDefinitions, v1alpha1.CustomResourceDefinitions{}) || len(csv.Spec.CustomResourceDefinitions.Owned) == 0 {
+		return nil, fmt.Errorf("no owned CustomResourceDefinitions specified on CSV %s, no custom resources to display", csvKey.String())
+	}
+
+	return csv.Spec.CustomResourceDefinitions.Owned, nil
+}
+
+// List takes in a list of CRDs and finds the associated CRs metadata on-cluster using the client-go meta client.
+// Note: List() can return a potentially unbounded list that callers may need to paginate.
+func (o *OperatorListCustomResources) List(ctx context.Context, crds []v1alpha1.CRDDescription) (result metav1.PartialObjectMetadataList, resErr error) {
+	for _, c := range crds {
+		// setup GVR
+		gvr := schema.GroupVersionResource{
+			Group:    c.Name,
+			Version:  c.Version,
+			Resource: strings.ToLower(c.Kind) + "s", // is this always a valid assumption?
+		}
+
+		// TODO use namespaced MetaClient if CR is namespace-scoped
+		r, err := o.config.MetaClient.Resource(gvr).List(ctx, metav1.ListOptions{})
```

Comment:
Instead of using a client-go MetaClient, is it possible to use controller-runtime client's `List` function with `unstructured.UnstructuredList` or `metav1.PartialObjectMetadataList`?

See [`List` implementation in `controller-runtime`](https://github.com/kubernetes-sigs/controller-runtime/blob/11dfabf5e2e83cdb8cba28fedcda0ae5e10474f0/pkg/client/client.go#L215-L225).